src/store_client.cc

```diff
@@ -491,34 +512,40 @@ store_client::fileRead()
               copyInto.data,
               copyInto.length,
               copyInto.offset + mem->swap_hdr_sz,
-              mem->swap_hdr_sz == 0 ? storeClientReadHeader
-              : storeClientReadBody,
+              expectHeader() ? storeClientReadHeader : storeClientReadBody,
```

Comment:
AFAICT, the old decision was about getting the so called "swapout header" and then reading the rest of the stored data (naturally called "body" in this context). The existing readHeader() and readBody() methods are also about that divide. However, the new undocumented expectHeader() is probably about the HTTP header. It looks out of place here.

Reply:
You are right, but your statements do not invalidate or address my concern -- the new condition "works", but it is either confusing (some readers will start thinking about wrong headers) or looks odd (if the reader knows exactly what is going on in those methods). I will resolve this request but may open another one, to attack the same problem from a different angle and in sync with the current code.